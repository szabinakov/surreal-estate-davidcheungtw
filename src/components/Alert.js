import React from "react";
import PropTypes from "prop-types";
import "../styles/alert.css";

const Alert = ({ message, isSuccess }) => {
  if (!message) return null;
  return (
    <div
      className={`${isSuccess ? "alert alert-success" : "alert alert-fail"}`}
    >
      {message}
    </div>
  );
};

export default Alert;

Alert.propTypes = {
  message: PropTypes.string.isRequired,
  isSuccess: PropTypes.bool.isRequired,
};
