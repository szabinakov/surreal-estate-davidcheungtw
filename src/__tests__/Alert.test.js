import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Alert from "../components/Alert";

describe("Alert", () => {
  const validProps1 = {
    message: "Error!",
    isSuccess: false,
  };
  const validProps2 = {
    message: "Success!!!",
    isSuccess: true,
  };

  it("Renders as expected", () => {
    const rendered = renderer.create(
      <Alert message={validProps1.message} isSuccess={validProps1.isSuccess} />
    );

    expect(rendered).toMatchSnapshot();
  });

  it("Renders with error message as expected", () => {
    const { asFragment } = render(
      <Alert message={validProps1.message} isSuccess={validProps1.isSuccess} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Assert error message is present", () => {
    render(
      <Alert message={validProps1.message} isSuccess={validProps1.isSuccess} />
    );

    expect(screen.getByText(/Error!/i)).toBeTruthy();
    expect(screen.getByText(/Error!/i)).toBeInstanceOf(HTMLDivElement);
    expect(screen.getByText(/Error!/i)).toHaveClass("alert-fail");
  });

  it("Renders with success message as expected", () => {
    const { asFragment } = render(
      <Alert message={validProps2.message} isSuccess={validProps2.isSuccess} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it("Assert success message is present", () => {
    render(
      <Alert message={validProps2.message} isSuccess={validProps2.isSuccess} />
    );

    expect(screen.getByText(/Success!!!/i)).toBeTruthy();
    expect(screen.getByText(/Success!!!/i)).toBeInstanceOf(HTMLDivElement);
    expect(screen.getByText(/Success!!!/i)).toHaveClass("alert-success");
  });
});
