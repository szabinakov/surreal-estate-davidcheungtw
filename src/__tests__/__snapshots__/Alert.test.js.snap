// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Renders as expected 1`] = `
<div
  className="alert alert-fail"
>
  Error!
</div>
`;

exports[`Alert Renders with error message as expected 1`] = `
<DocumentFragment>
  <div
    class="alert alert-fail"
  >
    Error!
  </div>
</DocumentFragment>
`;

exports[`Alert Renders with success message as expected 1`] = `
<DocumentFragment>
  <div
    class="alert alert-success"
  >
    Success!!!
  </div>
</DocumentFragment>
`;
